lib/Xmds/Soap.php

```diff
@@ -427,6 +429,47 @@ protected function doRequiredFiles($serverKey, $hardwareKey, $httpDownloads)
             return new \SoapFault('Sender', 'Unable to get a list of layouts');
         }
 
+        // workout if any of the layouts we have in our list has Actions pointing to another Layout.
+        foreach ($layouts as $layoutId) {
```

Comment:
Looking at this, do we need to have a method in layout factory which just returns the ID? I'm a bit concerned we'll end up running getByCode potentially a bunch of times? which then runs the full query.

Would it actually be possible to have a `$layout->getActionLayoutIds()` method which just runs some SQL to get a bunch of IDs? That would be super efficient by comparison.